scripts/views/mapboxView.js

```diff
@@ -373,17 +441,17 @@
   mapboxView.toggleBorders = function () {
     $('.border-toggle').removeClass('active');
     $(this).addClass('active');
-
+    mapboxView.removeSelections();
+    mapboxView.removeListeners();
     if (this.id === 'show-federal-borders') {
+      mapboxView.removeSelections();
       map.setLayoutProperty('district_border', 'visibility', 'visible');
-
-      map.setLayoutProperty('states-house-districts', 'visibility', 'none');
-      map.setLayoutProperty('states-senate-districts', 'visibility', 'none');
+      mapboxView.addDistrictListener();
     } else if (this.id === 'show-state-borders') {
-      map.setLayoutProperty('district_border', 'visibility', 'none');
-
+      mapboxView.removeSelections();
```

Comment:
same here, just do it before the if block

Reply:
k